### PR TITLE
[office-js][office-js-preview] (Outlook) Document Mailbox 1.14

### PR DESCRIPTION
**Pull request by samantharamon**
- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:

- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: None.
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

### CODE REVIEW

File: types/office-js-preview/index.d.ts [hunk 1]
```diff
@@ -539,7 +540,9 @@ declare namespace Office {
          */
         AppointmentTimeChanged,
         /**
-         * Occurs when an attachment is added to or removed from an item. **Important**: This event can only be handled in a task pane. It isn't supported by function commands.
+         * Occurs when an attachment is added to or removed from an item.
+         *
+         * **Important**: This event can only be handled in a task pane. It isn't supported by function commands.
```
Comment (Rick-Kirkham) [inline types/office-js-preview/index.d.ts:545]: Isn't this true of all events? I don't see how a function that's triggered by a ribbon button can handle an event. Or is this trying to say that a function command can't _register_ a handler for an event?

File: types/office-js-preview/index.d.ts [hunk 2]
```diff
@@ -8437,44 +8451,36 @@ declare namespace Office {
...
          * **{@link https://learn.microsoft.com/office/dev/add-ins/outlook/outlook-add-ins-overview#extension-points | Applicable Outlook mode}**: Compose or Read
          *
          * **Important**:
          *
-         * - This property is currently in preview in Outlook on Windows. To test it in your add-in, you must install Version 2401 (Build 17228.20000) or later.
-         * Then, join the {@link https://insider.microsoft365.com/join/windows | Microsoft 365 Insider program} and select the **Beta Channel** option to access
-         * Office beta builds.
-         *
-         * - The URL returned points to the location of the JavaScript file that Outlook on Windows uses to handle event-based activation
+         * - The URL returned points to the location of the JavaScript file that classic Outlook on Windows uses to handle event-based activation
```
Comment (Rick-Kirkham) [inline types/office-js-preview/index.d.ts:8474]:
```suggestion
         * - The URL that is returned points to the location of the JavaScript file that classic Outlook on Windows uses to handle event-based activation
```

**Review by Rick-Kirkham — CHANGES_REQUESTED** [review]

File: types/office-js-preview/index.d.ts (hunk 1 above)
Comment (samantharamon) [reply types/office-js-preview/index.d.ts:545]: It refers to the latter. I'll update the wording to make this clearer.

**Review by Rick-Kirkham — APPROVED** [review]

**Review by jakebailey — APPROVED** [review]

File: types/office-js-preview/index.d.ts [hunk 3]
```diff
@@ -20551,103 +20515,91 @@ declare namespace Office {
...
          * To add an optional text box to the preprocessing dialog of your spam-reporting add-in, you must configure the
-         * {@link https://learn.microsoft.com/javascript/api/manifest/preprocessingdialog?view=outlook-js-preview&preserve-view=true#child-elements | FreeTextLabel}
+         * {@link https://learn.microsoft.com/javascript/api/manifest/preprocessingdialog#child-elements | FreeTextLabel}
          * element in the manifest of your add-in.
          *
          * To learn more about how to develop a spam-reporting add-in, see
-         * {@link https://learn.microsoft.com/office/dev/add-ins/outlook/spam-reporting | Implement an integrated spam-reporting add-in (preview)}.
-         *
-         * @beta
+         * {@link https://learn.microsoft.com/office/dev/add-ins/outlook/spam-reporting | Implement an integrated spam-reporting add-in }.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js-preview/index.d.ts:20557]:
```suggestion
         * {@link https://learn.microsoft.com/office/dev/add-ins/outlook/spam-reporting | Implement an integrated spam-reporting add-in}.
```

File: types/office-js-preview/index.d.ts [hunk 4]
```diff
@@ -20551,103 +20515,91 @@ declare namespace Office {
...
          * The order of the booleans in the array corresponds to the order of the reporting options specified in the
-         * {@link https://learn.microsoft.com/javascript/api/manifest/reportingoptions?view=outlook-js-preview&preserve-view=true | ReportingOptions}
+         * {@link https://learn.microsoft.com/javascript/api/manifest/reportingoptions | ReportingOptions}
          * element of your add-in's manifest.
          *
          * To learn more about how to develop a spam-reporting add-in, see
-         * {@link https://learn.microsoft.com/office/dev/add-ins/outlook/spam-reporting | Implement an integrated spam-reporting add-in (preview)}.
-         *
-         * @beta
+         * {@link https://learn.microsoft.com/office/dev/add-ins/outlook/spam-reporting | Implement an integrated spam-reporting add-in }.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js-preview/index.d.ts:20571]:
```suggestion
         * {@link https://learn.microsoft.com/office/dev/add-ins/outlook/spam-reporting | Implement an integrated spam-reporting add-in}.
```

File: types/office-js-preview/index.d.ts [hunk 5]
```diff
@@ -20756,9 +20700,7 @@ declare namespace Office {
          * reported message. This applies even if `showPostProcessingDialog` is configured. However, depending on how you configured the `moveItemTo` property in the
          * `event.completed` call, a notification is shown to signal when the reported message is deleted or moved to another folder in the mailbox. To learn more, see
          * the "Review feature behavior and limitations" section of
-         * {@link https://learn.microsoft.com/office/dev/add-ins/outlook/spam-reporting#review-feature-behavior-and-limitations | Implement an integrated spam-reporting add-in (preview)}.
-         *
-         * @beta
+         * {@link https://learn.microsoft.com/office/dev/add-ins/outlook/spam-reporting#review-feature-behavior-and-limitations | Implement an integrated spam-reporting add-in }.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js-preview/index.d.ts:20703]:
```suggestion
         * {@link https://learn.microsoft.com/office/dev/add-ins/outlook/spam-reporting#review-feature-behavior-and-limitations | Implement an integrated spam-reporting add-in}.
```

File: types/office-js/index.d.ts [hunk 6]
```diff
@@ -8387,6 +8439,44 @@ declare namespace Office {
...
+         * **{@link https://learn.microsoft.com/office/dev/add-ins/outlook/understanding-outlook-add-in-permissions | Minimum permission level}**: **restricted**
+         *
+         * **{@link https://learn.microsoft.com/office/dev/add-ins/outlook/outlook-add-ins-overview#extension-points | Applicable Outlook mode}**: Compose or Read
+         *
+         * **Important**:
+         *
+         * - The URL that's returned points to the location of the JavaScript file that classic Outlook on Windows uses to handle event-based activation
+         * and integrated spam reporting. To learn more about these features, see
+         * {@link https://learn.microsoft.com/office/dev/add-ins/outlook/autolaunch | Configure your Outlook add-in for event-based activation} and
+         * {@link https://learn.microsoft.com/office/dev/add-ins/outlook/spam-reporting | Implement an integrated spam-reporting add-in }.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js/index.d.ts:8470]:
```suggestion
         * {@link https://learn.microsoft.com/office/dev/add-ins/outlook/spam-reporting | Implement an integrated spam-reporting add-in}.
```

File: types/office-js/index.d.ts [hunk 7]
```diff
@@ -8963,6 +9083,43 @@ declare namespace Office {
              */
             Dec = "dec"
         }
+        /**
+         * Specifies the folder to which a reported spam or phishing message is moved once it's processed by a spam-reporting add-in.
+         *
+         * To learn more about the integrated spam-reporting feature, see
+         * {@link https://learn.microsoft.com/office/dev/add-ins/outlook/spam-reporting | Implement an integrated spam-reporting add-in }.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js/index.d.ts:9090]:
```suggestion
         * {@link https://learn.microsoft.com/office/dev/add-ins/outlook/spam-reporting | Implement an integrated spam-reporting add-in}.
```

File: types/office-js/index.d.ts [hunk 8]
```diff
@@ -19456,6 +20164,195 @@ declare namespace Office {
...
+         *
+         * @remarks
+         * [Api set: Mailbox 1.14]
+         *
+         * To add an optional text box to the preprocessing dialog of your spam-reporting add-in, you must configure the
+         * {@link https://learn.microsoft.com/javascript/api/manifest/preprocessingdialog#child-elements | FreeTextLabel}
+         * element in the manifest of your add-in.
+         *
+         * To learn more about how to develop a spam-reporting add-in, see
+         * {@link https://learn.microsoft.com/office/dev/add-ins/outlook/spam-reporting | Implement an integrated spam-reporting add-in }.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js/index.d.ts:20207]:
```suggestion
         * {@link https://learn.microsoft.com/office/dev/add-ins/outlook/spam-reporting | Implement an integrated spam-reporting add-in}.
```

File: types/office-js/index.d.ts [hunk 9]
```diff
@@ -19456,6 +20164,195 @@ declare namespace Office {
...
+         *
+         * @remarks
+         * [Api set: Mailbox 1.14]
+         *
+         * The order of the booleans in the array corresponds to the order of the reporting options specified in the
+         * {@link https://learn.microsoft.com/javascript/api/manifest/reportingoptions | ReportingOptions}
+         * element of your add-in's manifest.
+         *
+         * To learn more about how to develop a spam-reporting add-in, see
+         * {@link https://learn.microsoft.com/office/dev/add-ins/outlook/spam-reporting | Implement an integrated spam-reporting add-in }.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js/index.d.ts:20221]:
```suggestion
         * {@link https://learn.microsoft.com/office/dev/add-ins/outlook/spam-reporting | Implement an integrated spam-reporting add-in}.
```

File: types/office-js/index.d.ts [hunk 10]
```diff
@@ -19456,6 +20164,195 @@ declare namespace Office {
...
+         *
+         * **{@link https://learn.microsoft.com/office/dev/add-ins/outlook/understanding-outlook-add-in-permissions | Minimum permission level (Outlook)}**: **read item**
+         *
+         * **{@link https://learn.microsoft.com/office/dev/add-ins/outlook/outlook-add-ins-overview#extension-points | Applicable Outlook mode}**: Message Read
+         *
+         * **Important**: In Outlook on the web or in new Outlook on Windows (preview), a post-processing dialog isn't shown once the add-in completes processing a
+         * reported message. This applies even if `showPostProcessingDialog` is configured. However, depending on how you configured the `moveItemTo` property in the
+         * `event.completed` call, a notification is shown to signal when the reported message is deleted or moved to another folder in the mailbox. To learn more, see
+         * the "Review feature behavior and limitations" section of
+         * {@link https://learn.microsoft.com/office/dev/add-ins/outlook/spam-reporting#review-feature-behavior-and-limitations | Implement an integrated spam-reporting add-in }.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js/index.d.ts:20353]:
```suggestion
         * {@link https://learn.microsoft.com/office/dev/add-ins/outlook/spam-reporting#review-feature-behavior-and-limitations | Implement an integrated spam-reporting add-in}.
```

**Review by ElizabethSamuel-MSFT — APPROVED** [review]

**Review by jakebailey — APPROVED** [review]

**Review by peterblazejewicz — APPROVED** [review]

**Comment by samantharamon** [comment]
Ready to merge

**Pull request merged**